packages/pwa-kit-runtime/src/utils/ssr-config.server.js

```diff
@@ -79,11 +102,38 @@ export const getConfig = (opts = {}) => {
 
     if (!config) {
         throw new Error(
-            `Application configuration not found!\nPossible configuration file locations:\n${searchPlaces.join(
-                '\n'
-            )}`
+            [
+                'Application configuration not found!',
+                'Possible configuration file locations:',
+                ...searchPlaces
+            ].join('\n')
         )
     }
 
     return config
 }
+
+/**
+ * Gets site specific configurations
+ *
+ * @param {string} [configFile] Optional `configFile` that containing the configution. If it
+ *      isn't specified, the `configFile` will resolved by resolution rules from the `getConfig`
+ * @param {object} opts
+ * @param {string} [opts.buildDirectory] Option path to the folder containing the configution.
+ * @param {string} [opts.siteId] Optional `siteId` that containing the configution. By default,
+ *      it is current site from `app.commerceAPI.parameters.siteId`
+ *
+ * @returns {object} The configution object
+ */
+export function getSiteConfig(configFile = '', {siteId = null, ...rest} = {}) {
```

Comment:
The runtime doesn't know what a "site" is and we shouldn't introduce this commerce concept into this package. See the comment below.

Reply:
This method is an optional helper. If we don't know the configuration in the config file, we can change the `siteId` argument to required and remove the default site fallback behavior _(or even remove this helper method from the package)_.
For more detail about the purpose of this method, please refer our conversation in the [origin issue](https://github.com/SalesforceCommerceCloud/pwa-kit/issues/1639#issuecomment-1902005404)